src/url-utils.ts

```diff
@@ -17,39 +17,40 @@
 
 import * as path from 'path';
 import * as url from 'url';
+import {ResolvedUrl, FileRelativeUrl, PackageRelativeUrl} from 'polymer-analyzer';
+
+// TODO(usergenic): parseUrl should be top-level export of polymer-analyzer.
 import {parseUrl} from 'polymer-analyzer/lib/core/utils';
 import constants from './constants';
 
 const sharedRelativeUrlProperties =
     ['protocol', 'slashes', 'auth', 'host', 'port', 'hostname'];
 
-/**
- * A string representing a URL.
- */
-export type UrlString = string;
+export type UrlString = ResolvedUrl | FileRelativeUrl | PackageRelativeUrl;
```

Comment:
?

Reply:
Some urls may be of any type in certain places.  I still want to indicate I expect them to be one of these in spirit.  For example, relativeUrl(source, target) needs source to be ResolvedUrl but target to be resolved or relative...  I guess I don't know enough about PackageRelativeUrl yet...  You caught me-- this UrlString change is a little uncooked.

Reply:
`PackageRelativeUrl` is pretty much now only used for urls passed in from APIs or the command line. They're just a way of expressing that this is a file that shouldn't be resolved relative to any other url. Hm, actually, we can express that in the type system can't we…

Reply:
https://github.com/Polymer/polymer-analyzer/pull/805 to clarify things a bit more using the type system